pilot/model/config_test.go

```diff
@@ -263,6 +264,36 @@ func TestResolveHostname(t *testing.T) {
 	}
 }
 
+func TestResolveFQDN(t *testing.T) {
+	cases := []struct {
+		name string
+		meta model.ConfigMeta
+		fqdn string
+	} {
+		{
+			name: "hello",
+			meta: model.ConfigMeta{Namespace: "default", Domain: "cluster.local"},
+			fqdn: "hello.default.svc.cluster.local",
+		},
+		{
+			name: "hello.default",
+			meta: model.ConfigMeta{Namespace: "default", Domain: "cluster.local"},
+			fqdn: "hello.default.svc.cluster.local",
+		},
+		{
+			name: "hello.default.svc.cluster.local",
+			meta: model.ConfigMeta{Namespace: "default", Domain: "cluster.local"},
+			fqdn: "hello.default.svc.cluster.local",
+		},
+	}
+
+	for _, test := range cases {
+		if got := model.ResolveFQDN(test.meta, test.name); got != test.fqdn {
```

Comment:
Here and in the other table driven tests in the PR, please wrap the test body in a `t.Run(func(t *testing.T){ ... })`. [It makes the test harder to screw up by e.g. using `t.Fatal` in the future.](https://blog.golang.org/subtests)